CMakeLists.txt

```diff
@@ -1,133 +1,93 @@
-        src/Circle.cpp
-        include/roboteam_utils/Circle.h
-        src/pid.cpp
-        include/roboteam_utils/pid.h
-        #src/Triangle.cpp
-        #include/roboteam_utils/Triangle.h
-        src/Tube.cpp
-        include/roboteam_utils/Tube.h
-        src/Time.cpp
-        include/roboteam_utils/Time.h
-        src/Random.cpp
-        include/roboteam_utils/Random.h
-        src/HalfLine.cpp
-        include/roboteam_utils/HalfLine.h
-        src/Grid.cpp
-        include/roboteam_utils/Grid.h
-        src/Shadow.cpp
-        include/roboteam_utils/Shadow.h
-        include/roboteam_utils/containers/circular_buffer.h
+        ${ROBOTEAM_UTILS_SRC}
         )
 
 
 target_include_directories(roboteam_utils
-        INTERFACE include
         PRIVATE include/roboteam_utils
-        PRIVATE src
+        INTERFACE include
         )
 
 target_link_libraries(roboteam_utils
         PUBLIC roboteam_proto
-        PUBLIC networking
+        PUBLIC utils_networking_lib
         )
 
-target_compile_options(roboteam_utils PRIVATE
-        -Wall
-        -Wextra
-        -pedantic
-        -Werror
-        -Wno-unused-function
-        -Wno-deprecated-declarations
-        -Wno-comment
-        -Wsign-compare
-        -g
-        )
-set_target_properties(roboteam_utils PROPERTIES
-        CMAKE_CXX_STANDARD 20
-        CMAKE_CXX_STANDARD_REQUIRED 1
-        )
 add_executable(utils_tests
-        test/main.cpp
-        test/AngleTest.cpp
-        test/ArcTest.cpp
-        #test/BiMapTest.cpp
-        test/LineTest.cpp
-        test/LineIntersectionTest.cpp
-        test/LineProjectionTest.cpp
-        test/CircleTest.cpp
-        test/normaliseTest.cpp
-        test/PolygonTest.cpp
-        test/PositionTest.cpp
-        test/Vector2Test.cpp
-        test/RectangleTest.cpp
-        test/HungarianTest.cpp
-        #test/TriangleTest.cpp
-        test/TubeTest.cpp
-        test/TimeTest.cpp
-        test/RandomTest.cpp
-        test/StateMachineTest.cpp
-        test/GridTest.cpp
-        test/StaticVectorTest.cpp
-        test/ShadowTest.cpp
-        test/CircularBufferTest.cpp
-        test/PubSubTest.cpp
+        ${ROBOTEAM_UTILS_TEST_SRC}
         )
 
 target_link_libraries(utils_tests
         PRIVATE roboteam_utils
         PRIVATE gtest
         )
-if(CODE_COVERAGE)
-    target_add_coverage_flags(roboteam_utils)
-    target_add_coverage_flags(networking)
-    target_add_coverage_flags(utils_tests)
-    target_link_libraries(utils_tests
-            PRIVATE --coverage)
 
-endif(CODE_COVERAGE)
+# -- NETWORKING --
+
+add_library(utils_networking_lib STATIC
+        src/networking/Channel.cpp
+        src/networking/Channels.cpp
+        src/networking/Module.cpp
+        )
+
+target_include_directories(utils_networking_lib
+        PUBLIC include
+        PRIVATE include/networking
+        )
+
+target_link_libraries(utils_networking_lib
+        PUBLIC stx
+        PUBLIC roboteam_proto
+        PUBLIC zmqpp
+        )
```

Comment:
why is this no longer in its own directory?

Reply:
The lib is so small I wasn't sure why it's even a separate target. I didn't like how there was both a top-level and nested src/ + include/. Since it was tiny, I just coalesced everything into a single top level source and include directory.